Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp

```diff
@@ -141,320 +120,474 @@ namespace AzToolsFramework
+        {
+            return false;
+        }
+
+        if (!m_localFileIO->Exists(directory.c_str()) || !m_localFileIO->IsDirectory(directory.c_str()))
+        {
+            AZ_Error(
+                s_traceName, false, "Working directory '%s' is not a directory or doesn't exist!", directory.c_str());
+            return false;
+        }
+
+        if (!file.empty())
+        {
+            auto filePath = AZ::IO::Path{ directory } / file;
+            if (!m_localFileIO->Exists(filePath.c_str()) || m_localFileIO->IsDirectory(filePath.c_str()))
             {
-                AZ::TickBus::QueueFunction(respCallback, (exitCode == static_cast<AZ::u32>(SevenZipExitCode::NoError)), AZStd::move(consoleOutput));
+                AZ_Error(s_traceName, false, "File list '%s' is a directory or doesn't exist!", filePath.c_str());
+                return false;
             }
+        }
 
-            if (!taskHandle.IsNull())
+        return true;
+    }
+
+    bool ArchiveComponent::CheckParamsForExtract(const AZStd::string& archive, const AZStd::string& directory)
+    {
+        if (!m_localFileIO || !m_archive)
+        {
+            return false;
+        }
+
+        if (!m_localFileIO->Exists(archive.c_str()))
+        {
+            AZ_Error(s_traceName, false, "Archive '%s' does not exist!", archive.c_str());
+            return false;
+        }
+
+        if (!m_localFileIO->Exists(directory.c_str()))
+        {
+            if (!m_localFileIO->CreatePath(directory.c_str()))
             {
-                AZStd::unique_lock<AZStd::mutex> lock(m_threadControlMutex);
-                ThreadInfo& tInfo = m_threadInfoMap[taskHandle];
-                tInfo.threads.erase(AZStd::this_thread::get_id());
-                m_cv.notify_all();
+                AZ_Error(s_traceName, false, "Failed to create destination directory '%s'", directory.c_str());
+                return false;
             }
-        };
-        if (!taskHandle.IsNull())
+        }
+
+        return true;
+    }
+
+    bool ArchiveComponent::CheckParamsForCreate(const AZStd::string& archive, const AZStd::string& directory)
+    {
+        if (!m_localFileIO || !m_archive)
+        {
+            return false;
+        }
+
+        if (m_localFileIO->Exists(archive.c_str()))
         {
-            AZStd::thread processThread(sevenZJob);
-            AZStd::unique_lock<AZStd::mutex> lock(m_threadControlMutex);
-            ThreadInfo& info = m_threadInfoMap[taskHandle];
-            m_cv.wait(lock, [&info, &processThread]() {
-                return info.threads.find(processThread.get_id()) != info.threads.end();
-            });
-            processThread.detach();
+            AZ_Error(s_traceName, false, "Archive file '%s' already exists, cannot create a new archive there!");
+            return false;
         }
-        else
+
+        if (!m_localFileIO->Exists(directory.c_str()) || !m_localFileIO->IsDirectory(directory.c_str()))
```

Comment:
repeat: The Exists check takes into account whether a file exists and a directory is a file, so the IsDirectory check isn't needed.

```suggestion
        if (!m_localFileIO->Exists(directory.c_str()))
```